### PR TITLE
Proxy/dev 115 rating

### PR DESCRIPTION
**Pull request by shev-titan**
- added rating strategy configuration and parameters with `rating-config.json`
- moved provider filtering responsibility to `rating.Rating` module, and its configuration to `rating-config.json`
- updated schema of model-config.json (kept old schema compatibility)
- added json schema for intellisense for model-config.json / rating-config.json

Enables configuration of weights change described here: https://github.com/Lumerin-protocol/Morpheus-Lumerin-Node/pull/266

### CODE REVIEW

File: proxy-router/internal/config/config.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ type Config struct {
 		StoreChatContext   *lib.Bool `env:"PROXY_STORE_CHAT_CONTEXT" flag:"proxy-store-chat-context" desc:"store chat context in the proxy storage"`
 		ForwardChatContext *lib.Bool `env:"PROXY_FORWARD_CHAT_CONTEXT" flag:"proxy-forward-chat-context" desc:"prepend whole stored message history to the prompt"`
 		ModelsConfigPath   string    `env:"MODELS_CONFIG_PATH" flag:"models-config-path" validate:"omitempty"`
-		ProviderAllowList  string    `env:"PROVIDER_ALLOW_LIST" flag:"provider-allow-list" validate:"omitempty" desc:"comma separated list of provider addresses allowed to open session with"`
```
Comment (abs2023) [inline proxy-router/internal/config/config.go:63]: @shev-titan please update /docs/proxy-router.all.env with this update 
Also, please update /docs/models-config.json.md based on these updates and 
add a /docs/rating-config.json.md helper document as well.

**Review by abs2023 — APPROVED** [review]

File: proxy-router/models-config.json.example [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-    "apiType": "openai",
-    "apiUrl": "http://localhost:8080/v1"
-  },
-  "0x0000000000000000000000000000000000000000000000000000000000000001": {
-    "modelName": "v1-5-pruned-emaonly.safetensors [d7049739]",
-    "apiType": "prodia-sd",
-    "apiUrl": "https://api.prodia.com/v1",
-    "apiKey": ""
-  }
+  "$schema": "./internal/config/models-config-schema.json",
```
[the remark this answers is not in the document]
Comment (shev-titan) [reply proxy-router/models-config.json.example:2]: It provides some validation, tried to enforce as much strictness as possible

File: proxy-router/rating-config.json
[the remark this answers is not in the document]
Comment (shev-titan) [reply proxy-router/rating-config.json:1]: if file is missing proxy won't start. We provide example for command-line users. The desktop users won't need to do anything.

File: proxy-router/internal/config/models-config-schema.json
[the remark this answers is not in the document]
Comment (shev-titan) [reply proxy-router/internal/config/models-config-schema.json:1]: extra entries will be ignored by default, or can be labeled as erroneous with more strict schema. I view json schemas as more of a documentation/ux instrument to help setup the configuration, not a source of truth. All configurations are validated during startup according to config.go

File: proxy-router/internal/config/config.go (hunk 1 above)
Comment (shev-titan) [reply proxy-router/internal/config/config.go:63]: done

**Pull request merged**